umm-spec-lib/src/cmr/umm_spec/migration/version/variable.clj

```diff
@@ -109,3 +109,30 @@
   ;; Migrate up to 1.3
   [context v & _]
   v)
+
+(defmethod interface/migrate-umm-version [:variable "1.4" "1.3"]
+  [context v & _]
+  ;; Migrate down to 1.3
+  (if-let [avg-comp-info (get-in v [:SizeEstimation :AverageCompressionInformation])]
+    (as-> v m
```

Comment:
This migration logic only works when both the ASCII and NetCDF-4 formats exist. You should add migration test cases for at least when one of the formats does not exist, and fix this migration code to make it work.

Reply:
What do you mean?  I only process v when AverageCompressionInformation exists in 1.4;  Then I only associate AvgCompressionRateASCII when "ASCII" format exists in AverageCompressionInformation;  And only associate AvgCompressionRateNetCDF4 when "NetCDF-4" format exists in AverageCompressionInformation. I've tested this function a few times. It works. I could add more test but the logic should be fine.

Reply:
I can't believe this. I tested a few times. and in the last minute I changed it to when-let. Thanks!